auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue

```diff
@@ -461,21 +577,274 @@ export default defineComponent({
+
+    const isModernizedEntity = (item: Business): boolean => {
+      const entityType = getEntityType(item)
+      const supportedEntityFlags = launchdarklyServices.getFlag(LDFlags.IaSupportedEntities)?.split(' ') || [] // move to somewhere factory?
+      console.log(supportedEntityFlags)
+      return supportedEntityFlags.includes(entityType)
+    }
+
+    const isColinEntity = (item: Business): boolean => { // not include ones modernized (feature flag)
+      const entityType = getEntityType(item)
+      // talk to Sev to see if there's better way to check colin
+      return (!isModernizedEntity(item) && (
+        entityType === CorpTypes.BC_COMPANY ||
+        entityType === CorpTypes.BC_CCC ||
+        entityType === CorpTypes.BC_ULC_COMPANY ||
+        entityType === CorpTypes.CCC_CONTINUE_IN ||
+        entityType === CorpTypes.BC_CORPORATION ||
+        entityType === CorpTypes.ULC_CONTINUE_IN ||
+        entityType === CorpTypes.EXTRA_PRO_REG ||
+        entityType === CorpTypes.LL_PARTNERSHIP ||
+        entityType === CorpTypes.LIMITED_CO ||
+        entityType === CorpTypes.LIM_PARTNERSHIP ||
+        entityType === CorpTypes.BC_UNLIMITED ||
+        entityType === CorpTypes.XPRO_LL_PARTNR ||
+        entityType === CorpTypes.XPRO_LIM_PARTNR
+      ))
+    }
+
+    const isSocieties = (item: Business): boolean => {
+      const entityType = getEntityType(item)
+      return (entityType === CorpTypes.CONT_IN_SOCIETY ||
+        entityType === CorpTypes.SOCIETY ||
+        entityType === CorpTypes.XPRO_SOCIETY)
+    }
+
+    const isOtherEntities = (item: Business): boolean => {
+      const entityType = getEntityType(item)
+      return (entityType === CorpTypes.FINANCIAL ||
+        entityType === CorpTypes.PRIVATE_ACT ||
+        entityType === CorpTypes.PARISHES)
+    }
+
+    const isAllowRestore = (item: Business): boolean => {
+      const entityType = getEntityType(item)
+      return (entityType === CorpTypes.BC_COMPANY ||
+        entityType === CorpTypes.BC_CCC ||
+        entityType === CorpTypes.BC_ULC_COMPANY ||
+        entityType === CorpTypes.COOP ||
+        entityType === CorpTypes.BENEFIT_COMPANY)
+    }
+
+    const getNrRequestDescription = (item: Business): string => { // can move to factory
+      const nrRequestActionCd = item.nameRequest?.requestActionCd
+      switch (nrRequestActionCd) {
+        case NrRequestActionCodes.AMALGAMATE:
+          return 'Amalgamate Now'
+        case NrRequestActionCodes.CONVERSION:
+          return 'Alter Now'
+        case NrRequestActionCodes.CHANGE_NAME:
+          return 'Change Name Now'
+        case NrRequestActionCodes.MOVE:
+          return 'Continue In Now'
+        case NrRequestActionCodes.NEW_BUSINESS: {
+          if (isOtherEntities(item)) {
+            return 'Download Form'
+          }
+          return 'Register Now'
+        }
+        case NrRequestActionCodes.RESTORE: {
+          return isAllowRestore(item) ? 'Restore Now' : 'Reinstate Now'
```

Comment:
Did you confirm with UX team that BC/CC/ULC/CP/BEN show the first label and the others show the second label?

I thought the logic was "if allow restore then show button else show something else"...?

Reply:
Yes, confirmed with Janis.

Screenshots from RC for reference:
![image](https://github.com/bcgov/sbc-auth/assets/60866283/7f542501-883e-4c81-920d-571fd88a9498)
![image](https://github.com/bcgov/sbc-auth/assets/60866283/377ed517-f07b-4150-b17c-a1c757c2dbcd)

I'll think of another name for this function. It needs better naming. :point_right: `isAllow...` is not accurate.

Reply:
Renaming the function is a good idea.

Does this action open the business' dashboard (Filings UI)?

Do we need a feature flag to enable/disable the action (else redirect to COLIN), for businesses we currently don't allow to restore/reinstate?

Reply:
> Do we need a feature flag to enable/disable the action (else redirect to COLIN), for businesses we currently don't allow to restore/reinstate?

I'm thinking of this. Will restore/reinstate be implemented before this branch merges into main? If so, I think no need to add a new FF. Now for button `Restore Now` and `Reinstate Now`, it will go to business dashboard if the business of NR is affiliated. Otherwise, it displays `Business unavailable` dialog.

Reply:
We have implemented restoration filings for some entity types only, but not fully released it to Prod. So, yes, a FF is needed here.

Maybe confirm with Janis or Mihai what should happen if there is a REH NR for an entity type not included in the FF.

Reply:
Resolved in #2433 .